strawberryfields/ops.py

```diff
@@ -725,66 +694,44 @@ def apply(self, reg, backend, **kwargs):
-        # evaluate the rest of the Parameters, restore the originals later
-        self.p = [z] + [x.evaluate() for x in self.p[1:]]
+        original_p0 = self.p[0]  # store the original Parameter
+        self.p[0] = z
 
         # convert RegRefs back to indices for the backend API
         temp = [rr.ind for rr in reg]
         # call the child class specialized _apply method
         self._apply(temp, backend, **kwargs)
-        self.p = original_p  # restore original unevaluated Parameter instances
+        self.p[0] = original_p0  # restore the original Parameter instance
 
     def merge(self, other):
         if not self.__class__ == other.__class__:
             raise MergeFailure('Not the same gate family.')
 
-        if len(self._extra_deps)+len(other._extra_deps) == 0:
-            # no extra dependencies <=> no RegRefTransform parameters,
-            # with which we cannot do arithmetic at the moment
-
-            # gates can be merged if they are the same class and share all the other parameters
-            if self.p[1:] == other.p[1:]:
-                # make sure the gates have the same dagger flag, if not, invert the second p[0]
-                if self.dagger == other.dagger:
-                    temp = other.p[0]
-                else:
-                    temp = -other.p[0]
-                # now we can add up the parameters and keep self.dagger
-                p0 = self.p[0] + temp
-                if p0 == 0:
-                    return None  # identity gate
-
-                # return a copy
-                # HACK: some of the subclass constructors only take a single parameter,
-                # some take two, none take three
-                if len(self.p) == 1:
-                    temp = self.__class__(p0)
-                else:
-                    temp = self.__class__(p0, *self.p[1:])
-                # NOTE deepcopy would make copies of RegRefs inside a possible
-                # RegRefTransformation parameter, RegRefs must not be copied.
-                # OTOH copy results in temp having the same p list as self,
-                # which we would modify below.
-                #temp = copy.copy(self)
-                #temp.p[0] = p0
-                temp.dagger = self.dagger
-                return temp
-
-        else:
-            # gates have RegRefTransform parameters:
-            # without knowing anything more specific about the gates, we
-            # can only merge them if they are each others' inverses
-            if self.p == other.p and self.dagger != other.dagger:
-                return None
+        # gates can be merged if they are the same class and share all the other parameters
+        if self.p[1:] == other.p[1:]:
+            # make sure the gates have the same dagger flag, if not, invert the second p[0]
+            if self.dagger == other.dagger:
+                temp = other.p[0]
+            else:
+                temp = -other.p[0]
```

Comment:
I'm a bit confused by the situation here. Is this because this method gets called like `G.merge(H)`?

Reply:
Yep